themes/bootstrap3/js/searchbox_controls.js

```diff
@@ -0,0 +1,143 @@
+      });
+    }
+  }
+
+  function _hideKeyboard() {
+    _keyboard.setOptions({
+      theme: _defaultTheme
+    });
+  }
+
+  function _onChange(input){
+    _textInput.value = input;
+    if (_textInput.value === '') {
+      _resetButton.hide();
+    } else {
+      _resetButton.show();
+    }
+    if ( typeof _keyboard !== 'undefined') {
+      _keyboard.setInput(input);
+    }
+    _textInput.dispatchEvent(new Event('input'));
+  }
+
+  function _onKeyPress(button){
+    if (button === "{shift}" || button === "{lock}") {
+      let currentLayoutType = _keyboard.options.layoutName;
+      _keyboard.setOptions({
+        layoutName: currentLayoutType === "default" ? "shift" : "default"
+      });
+    }
+
+    if (button === "{enter}") {
+      document.getElementById("searchForm").submit();
+    }
+  }
+
+  function _updateKeyboardLayout(layoutName) {
+    $('.keyboard-selection-item').each(function deactivateItems() {
+      $(this).parent().removeClass("active");
+    });
+    $(".keyboard-selection-item[data-value='" + layoutName + "']").parent().addClass("active");
+    window.Cookies.set("keyboard", layoutName);
+    if (layoutName === "none") {
+      _enabled = false;
+      _hideKeyboard();
+    } else {
+      _enabled = true;
+      const keyboardLayout = new _KeyboardLayoutClass().get(layoutName);
+      _keyboard.setOptions({layout: keyboardLayout.layout});
+    }
+  }
+
+  function _initKeyboard(){
+    _KeyboardClass = window.SimpleKeyboard.default;
+    _KeyboardLayoutClass = window.SimpleKeyboardLayouts.default;
+
+    $('.keyboard-selection-item').on("click", function updateLayoutOnClick(){
+      _updateKeyboardLayout($(this).data("value"));
+    });
+
+    _textInput.addEventListener("focus", () => {
+      _showKeyboard();
+    });
+    _textInput.addEventListener("click", () => {
+      _showKeyboard();
+    });
+    document.addEventListener("click", (event) => {
+      if (
+        _keyboard.options.theme.includes("show-keyboard") &&
+        !event.target.className.includes("searchForm_lookfor") &&
+        !event.target.className.includes("hg-button") &&
+        !event.target.className.includes("hg-row") &&
+        !event.target.className.includes("simple-keyboard") &&
+        !event.target.className.includes("searchForm-reset") &&
+        !event.target.parentNode.className.includes("searchForm-reset")
+      ) {
+        _hideKeyboard();
+      }
+    });
+
+    _keyboard = new _KeyboardClass(
+      {
+        onChange: input => _onChange(input),
+        onKeyPress: button => _onKeyPress(button),
+        display: _display,
+        syncInstanceInputs: true,
+        mergeDisplay: true,
+        physicalKeyboardHighlight: true
+      });
+
+    let layout = window.Cookies.get("keyboard");
+    if (layout == null) {
+      layout = "none";
+    }
+    _updateKeyboardLayout(layout);
+  }
+
+  function init(){
+    _textInput = document.querySelector(".searchForm_lookfor");
```

Comment:
Is there a reason to use the class `.searchForm_lookfor` for the text input but the id `#searchForm-reset` for the reset button? Might it be safer to use identifiers in both cases to be sure that only one element is selected? I'm concerned that strange and unpredictable things might occur if we ever had a scenario where multiple elements matching the `.searchForm_lookfor` class were on screen at once. (And if we need to support a multi-input scenario, we would then need to revise the reset button logic).

Reply:
Changed it to "document.getElementById" in both cases as in #3064. 

I thought about using this also for advanced searches where multiple input fields are present. This would require refactoring this anyway. But that is something for a followup PR.